avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -935,7 +939,20 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
           int precision = precisionString == null ? CONNECT_AVRO_DECIMAL_PRECISION_DEFAULT :
               Integer.parseInt(precisionString);
           int scale = scaleString == null ? 0 : Integer.parseInt(scaleString);
-          org.apache.avro.LogicalTypes.decimal(precision, scale).addToSchema(baseSchema);
+          if (scale < 0 || scale > precision) {
```

Comment:
How about a comment here to describe why this scenario needs special logic and how that logic deviates from using `LogicalTypes.decimal(precision, scale).addToSchema(...)`?

Reply:
Ack, addressed